src/ProtocolFeeController.sol

```diff
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright (C) 2024 PancakeSwap
+pragma solidity 0.8.26;
+
+import {Ownable} from "./base/Ownable.sol";
+import {PoolKey} from "./types/PoolKey.sol";
+import {PoolId} from "./types/PoolId.sol";
+import {Currency} from "./types/Currency.sol";
+import {LPFeeLibrary} from "./libraries/LPFeeLibrary.sol";
+import {IProtocolFeeController} from "./interfaces/IProtocolFeeController.sol";
+import {IProtocolFees} from "./interfaces/IProtocolFees.sol";
+import {ProtocolFeeLibrary} from "./libraries/ProtocolFeeLibrary.sol";
+
+/// @notice ProtocolFeeController for both Pool type
+contract ProtocolFeeController is IProtocolFeeController, Ownable {
```

Comment:
i think we'll have feedback to use `Ownable2Step` here like vault -- so probably just use `Ownable2Step` to make it easier 😆